cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactVersion.java

```diff
@@ -30,11 +30,12 @@
  * parse version string in format of [major].[minor].[fix](-|.)[suffix]
  * }
  * </pre>
+ * Note that the suffix can not have numerical values in it.
  */
 @Beta
 public final class ArtifactVersion implements Comparable<ArtifactVersion> {
 
-  private static final Pattern PATTERN = Pattern.compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-](.*))?");
+  private static final Pattern PATTERN = Pattern.compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-]([^\\d]*))?$");
```

Comment:
Why need to exclude number from suffix? I think the right logic should be using the old pattern, but then keep calling `matcher.find()` while it returns true until `matcher.hitEnd()` return true. Something like this:

```java
boolean matches = false;
if (matchSuffix) {
  while (!matches && matcher.find()) {
    matches = matcher.hitEnd();
  }
} else {
  matches = matcher.matches();
}
```